html_table_parser/parser.py

```diff
@@ -37,9 +37,13 @@ def handle_starttag(self, tag, attrs):
 
     def handle_data(self, data):
         """ This is where we save content to a cell """
-        if self._in_td ^ self._in_th:
+        if self._in_td or self._in_th:
             self._current_cell.append(data.strip())
 
+    def handle_charref(self, name):
+        """ Handle HTML encoded characters """
+        self.handle_data(self.unescape('&#{};'.format(name)))
```

Comment:
Is there any reason for that you interpret content of a table cell? I think it's not the purpose of an HTML table parser to do so.